### PR TITLE
Validating check service exists using either service type or service name

### PR DESCRIPTION
**Pull request by Ameenasuhani**
Validating check service exists using either service type or service name

### CODE REVIEW

File: ceph/ceph_admin/orch.py [hunk 1]
```diff
@@ -77,25 +82,34 @@ def check_service_exists(
...
-            "args": {"service_type": service_type, "refresh": True},
+            "args": {"refresh": True},
         }
 
+        if service_name:
+            svc = service_name
+            arg = {"service_name": service_name}
+        elif service_type:
+            svc = service_type
+            arg = {"service_type": service_type}
```
Comment (psathyan) [inline ceph/ceph_admin/orch.py:93]: This makes it either or... instead we can
```
if service_name:
  check_status_dict["args"]["service_name"] = service_name

if service_type:
  check_status_dict["args"["service_type"] = service_type
```

File: ceph/ceph_admin/orch.py [hunk 2]
```diff
@@ -77,25 +82,34 @@ def check_service_exists(
...
+
         while end_time > datetime.now():
             sleep(interval)
             out, err = self.ls(check_status_dict)
             out = loads(out)[0]
             running = out["status"]["running"]
             count = out["status"]["size"]
 
-            LOG.info("%s/%s %s daemon(s) up... retrying", running, count, service_type)
+            LOG.info("%s/%s %s daemon(s) up... retrying", running, count, svc)
```
Comment (psathyan) [inline ceph/ceph_admin/orch.py:104]:
```
LOG.info(f"{running}/{count} {service_name if service_name else service_type} up... retrying")
```

File: ceph/ceph_admin/orch.py [hunk 3]
```diff
@@ -303,8 +315,8 @@ def verify_status(self, op) -> None:
             return True
         return False
 
-    def validate_spec_services(self, specs) -> None:
+    def validate_spec_services(self, specs) -> bool:
         LOG.info("Validating spec services")
         for spec in specs:
-            self.check_service_exists(service_type=spec["service_type"])
+            self.check_service_exists(service_name=spec["service_type"])
```
Comment (psathyan) [inline ceph/ceph_admin/orch.py:321]: I believe this should be `service_type`

**Comment by Ameenasuhani** [comment]
Rgw tier-0: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-0ZYUKO/
Apply spec: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-EY03WN/
Cephadm tier-0: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ZK6JDU/

**Comment by Ameenasuhani** [comment]
@psathyan Done

**Pull request merged**